### PR TITLE
Feature/mvp 1/todo creation

### PR DESCRIPTION
**Pull request by aniketrathi**
## Description
Added an endpoint to create a new TODO item.

URI: POST /todos
Request body: { title: … }
Response HTTP Status: 201
Response body: { id: …, title: … }

## Database schema changes
New model Todo added.
Schema:
{
title: { type: String, required: true }
},
{
collection: 'todos',
timestamps: {
createdAt: 'createdAt',
updatedAt: 'updatedAt',
},
}
### Automated test cases added
- Should create a new todo if valid string is passed as "title" in body, it should return 201
- Should give 400 if empty string is passed as "title" in body
- Should give 400 if non-string value is passed as "title" in body

### Manual test cases run
- None

### CODE REVIEW

File: src/storage/mongoose/mongo-store.ts [hunk 1]
```diff
@@ -8,9 +7,10 @@ import {
   Mongoose,
   Types,
 } from 'mongoose';
-import { Account, BaseModel, ModelFactory } from '@models';
+import { Todo,BaseModel, ModelFactory } from '@models';
```
Comment (balwant-jtc) [inline src/storage/mongoose/mongo-store.ts]: Todo,BaseModel => Todo, BaseModel

Comment (aniketrathi) [reply src/storage/mongoose/mongo-store.ts]: ok

File: src/models/todo.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import { BaseModel } from "./base-model";
+import { LooseObject } from "@typings";
+
+export class Todo extends BaseModel {
+  title: string;
+
+  constructor(json?: any) {
```
Comment (balwant-jtc) [inline src/models/todo.ts]: why json?

Comment (balwant-jtc) [reply src/models/todo.ts]: Use LooseObject instead of any

Comment (aniketrathi) [reply src/models/todo.ts]: done

File: src/models/todo.ts [hunk 3]
```diff
@@ -0,0 +1,20 @@
+import { BaseModel } from "./base-model";
+import { LooseObject } from "@typings";
+
+export class Todo extends BaseModel {
```
Comment (jjalan) [inline src/models/todo.ts]: Should we use `TodoItem` instead of `Todo` to make it a noun instead of a verb?

File: test/spec/todo/todo.spec.ts [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+    const res = await chai.request(expressApp).post("/todos").send({
+      title: "",
+    });
+    expect(res).to.have.status(400);
+    expect(res.body)
+      .to.have.nested.property("failures[0].message")
+      .to.equal("Please specify the valid title");
+  });
+
+  it("should return a validation error if title is not a string", async () => {
```
Comment (jjalan) [inline test/spec/todo/todo.spec.ts:48]: good

**Review by jjalan — APPROVED** [review]

**Review by jjalan — COMMENTED** [review]
Kindly make PR title user friendly. The current one (below) is no good.

```
Feature/mvp 1/todo creation
```

File: src/models/todo.ts (hunk 3 above)
Comment (aniketrathi) [reply src/models/todo.ts]: ya sure

**Pull request merged**